src/psyclone/parse/algorithm.py

```diff
@@ -594,9 +598,28 @@ def get_kernel(parse_tree, alg_filename):
             var_name = var_name.lower()
             arguments.append(Arg('indexed_variable', full_text, var_name))
         elif isinstance(argument, Data_Ref):
+            # A structure dereference e.g. base%arg, base%arg(n)
             full_text = argument.tostr().lower()
             var_name = create_var_name(argument).lower()
             arguments.append(Arg('variable', full_text, var_name))
+        elif isinstance(argument, (Level_2_Unary_Expr, Add_Operand,
+                                   Parenthesis)):
+            # An expression e.g. -1, 1*n, ((1*n)/m). Note, for some
+            # reason Add_Operation represents binary expressions in
+            # fparser2.  Walk the tree to look for an argument.
+            found_arg = False
+            for statement in walk_ast([argument]):
```

Comment:
You could do `walk_ast([argument], [Name])` to just get all nodes that are instances of Name.

Reply:
Great. I've simplified.